src/java/htsjdk/samtools/util/Tuple.java

```diff
@@ -13,4 +13,28 @@ public Tuple(final A a, final B b) {
         this.a = a;
         this.b = b;
     }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        final Tuple<?, ?> tuple = (Tuple<?, ?>) o;
+
+        if (a != null ? !a.equals(tuple.a) : tuple.a != null) return false;
+        return !(b != null ? !b.equals(tuple.b) : tuple.b != null);
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = a != null ? a.hashCode() : 0;
+        result = 31 * result + (b != null ? b.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "[" + a.toString() + ", " + b.toString() + "]";
```

Comment:
So this will explode if there is a null still...  This is actually a weird case, because 
```
 return "[" + a+ ", " + b + "]";
```

will be just fine.  I never thought about it before, but there must be special syntactic sugar that checks nulls in string concatenations.